_posts/2022-02-15-label-noise-intro.md

```diff
@@ -0,0 +1,121 @@
+
+In this method, we first randomly retag x % of samples. Then we identify the major tag confusions (as shown in the Dataset section) and pick the top 5 noisy tags and increase the weights associated to these tags in the sampling function. Then we sample again - pick top 5 tags - repeat.
+
+![image info](../assets/images/label-noise-blog/random-sampling-weights-plot.png)
+
+We see an improvement over the baseline. We can capture around 60% of the total label noise by just tagging around 32% of the total dataset by this heuristic.
+
+## Datamaps
+We leverage the training artefacts from data maps to define a label score for each sample - as the Euclidean distance between (0,0) and (confidence, variability). Following the hypothesis of hard-to-learn regions, we expect noisy samples to have a lower label score.
+* **Threshold on label-score** \
+  Fixing a threshold on the label score means that all samples that score below it are considered label noise, and those that score above are considered clean. Assuming we do Human Retagging of all samples predicted as label noisy, fixing a threshold essentially fixes both the % of samples retagged and (given the clean tags-) label noise recall. Varying the threshold, we get a plot for our dataset:
+
+  ![image info](../assets/images/label-noise-blog/deja-vu-plot.png)
+  Looking at our metrics, we see an improvement in the partial recleaning process.
+  Lets read the above plots. Say we fix the threshold at 0.43 which means we would be retagging around 28% of our dataset. This corresponds to a label noise recall of 60%, giving us a resulting dataset with 5.2% label noise from 13%. (= 0.40*13).
+
+* **n-consecutive correct instances** \
+  Here, we will use the ordering of the label scores. Note that this is an extension of the original HTL hypothesis on Data Maps - which creates partitions based on only thresholds (on the training artefacts - confidence, variability). Our added assumption here, is that the ordering within the regions are also useful.
+  Based on this, we sort our samples by label score, and in ascending order. This means the noisy samples should be nearer to the top, and we base our heuristic on this. We start Human Retagging from the top of the sorted list of samples, and stop once we see N-consecutive clean samples. Note how different this is from anything we have done before - because we are combining the Human Retagging process into the label noise id one.\
+  Varying N, we get a plot for our dataset:
+
+  ![image info](../assets/images/label-noise-blog/deja-vu-n-consecutive.png)
+  Again, we see an improvement in the partial recleaning process. Lets read the above plots. Say we fix N at ~38, which means we would be retagging around 35% of our dataset. This corresponds to a corresponds to a label noise recall of 60%, which means we would capture and clean 76% of the label noise. Giving us a resulting dataset with 3.1% label noise (= (1-0.76)*0.13).
+
+
+
+## Cleanlab
```

Comment:
Can we make this heading level focus more on the method rather than the tool? E.g. here we can talk about confident learning as the heading that will then bring in cleanlab within it.